src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1678,9 +1660,9 @@ void CASESession::HandleSigma3b(HandleSigma3Work & work)
     CompressedFabricId unused;
     FabricId initiatorFabricId;
     P256PublicKey initiatorPublicKey;
-    SuccessOrExit(err = FabricTable::VerifyCredentials(work.initiatorNOC, work.initiatorICAC, work.fabricRCAC, work.validContext,
-                                                       unused, initiatorFabricId, work.initiatorNodeId, initiatorPublicKey));
-    VerifyOrExit(work.fabricId == initiatorFabricId, err = CHIP_ERROR_INVALID_CASE_PARAMETER);
+    SuccessOrExit(err = FabricTable::VerifyCredentials(data.initiatorNOC, data.initiatorICAC, data.fabricRCAC, data.validContext,
```

Comment:
As a followup, worth changing everything in this function to ReturnErrorOnFailure() and VerifyOrReturnError(), since the exit block doesn't clean anything up anymore?

Reply:
I was keeping it consistent with past code and other code in the file, but maybe that doesn't make sense any more.

Reply:
I can do that in a follow-up, issue #26280